src/Network/Broadcast/OutboundQueue.hs

```diff
@@ -312,14 +313,23 @@ type InFlight nid = Map nid (Map Precedence Int)
 -- trying again
 type Failures nid = Map nid (UTCTime, ReconsiderAfter)
 
-inFlightTo :: Ord nid => nid -> Lens' (InFlight nid) (Map Precedence Int)
+inFlightTo :: Ord nid => nid -> Getter (InFlight nid) (Map Precedence Int)
 inFlightTo nid = at nid . anon Map.empty Map.null
 
-inFlightWithPrec :: Ord nid => nid -> Precedence -> Lens' (InFlight nid) Int
+inFlightWithPrec :: Ord nid => nid -> Precedence -> Getter (InFlight nid) Int
 inFlightWithPrec nid prec = inFlightTo nid . at prec . anon 0 (== 0)
 
-inFlightFor :: Ord nid => Packet msg nid a -> Lens' (InFlight nid) Int
-inFlightFor Packet{..} = inFlightWithPrec packetDestId packetPrec
+-- | Given an update function and a `Packet`, set the `InFlight` to the new value
+-- calculated by the updater. In case no match can be found, this function is
+-- effectively a noop.
+setInFlightFor :: (MonadIO m, Ord nid)
+               => Packet msg nid a
+               -> (Int -> Int)
+               -> MVar (InFlight nid)
+               -> m ()
+setInFlightFor Packet{..} updater var = liftIO $ modifyMVar_ var $ \iflight -> do
+    let updateInner = Map.adjust updater packetPrec
+    return $ Map.adjust updateInner packetDestId iflight
```

Comment:
Minor nitpick: this could be more functional and less imperative in style:
```
setInFlightFor Packet{..} f var =
    liftIO $ modifyMVar_ var (return . update)
  where
    update :: InFlight nid -> InFlight nid
    update = Map.adjust (Map.adjust f packetPrec) packetDestId 
```

Reply:
ok!